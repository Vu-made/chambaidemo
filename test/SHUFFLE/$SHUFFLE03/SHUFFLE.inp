20 100
15 5
3 6
8 5
8 18
6 13
11 20
8 4
5 4
14 18
2 7
18 10
2 14
7 13
20 7
6 18
20 10
11 18
2 10
14 19
11 2
11 1
1 9
19 17
5 15
2 19
9 14
3 13
17 11
3 4
5 9
5 12
18 2
15 4
12 19
10 13
14 1
6 15
8 2
17 16
17 15
6 4
11 7
17 1
2 19
2 15
13 16
9 10
9 7
9 18
11 9
8 6
6 17
11 6
4 10
13 6
6 8
7 18
5 18
9 10
17 15
12 16
17 2
17 18
15 8
13 20
14 1
2 13
5 6
4 7
6 1
19 3
5 15
8 7
6 12
4 15
1 14
1 5
4 11
10 19
4 18
14 7
2 11
19 7
14 15
10 7
17 1
5 16
2 10
11 18
6 19
5 19
13 11
11 20
10 18
13 12
11 14
13 9
14 5
8 18
13 1
