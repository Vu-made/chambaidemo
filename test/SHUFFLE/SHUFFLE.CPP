#include <bits/stdc++.h>
#define N 100005
using namespace std;
int Next[N], Pre[N];
int n,x;
int main()
{
    freopen("SHUFFLE.INP","r",stdin);
    freopen("SHUFFLE.OUT","w",stdout);
    cin >> n >> x;
    Next[0] = 1; Pre[n+1] = n;
    for (int i=1;i<=n; i++)
    {
        Next[i] = i+1;
        Pre[i] = i-1;
    }
    for (int i,j,k=1;k <= x;k++)
    {
        cin >> i >> j;
        Next[Pre[i]] = Next[i];
        Pre[Next[i]] = Pre[i];
        Next[i] = j;
        Pre[i] = Pre[j];
        Next[Pre[j]] = i;
        Pre[j] = i;
    }
    for (int p=Next[0]; p<n+1; p=Next[p])
        cout << p << " ";
    return 0;
}
