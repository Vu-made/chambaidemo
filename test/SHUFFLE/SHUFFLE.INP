9 3
8 2
4 7
1 9
