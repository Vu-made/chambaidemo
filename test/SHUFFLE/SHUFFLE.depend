# depslib dependency file v1.0
1602671316 source:c:\users\admin\desktop\bgdv\bt\thuat_toan\array1d_3\shuffle\shuffle.cpp
	<bits/stdc++.h>

