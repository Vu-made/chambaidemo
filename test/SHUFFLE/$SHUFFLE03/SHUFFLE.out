3 17 13 1 11 15 20 9 12 16 6 14 5 19 2 7 4 10 8 18 